samza-core/src/main/java/org/apache/samza/zk/BarrierForVersionUpgrade.java

```diff
@@ -19,29 +19,23 @@
 
 package org.apache.samza.zk;
 
-import java.util.List;
-
-
 /**
  * Interface for a barrier - to allow synchronization between different processors to switch to a newly published
  * JobModel.
  */
 public interface BarrierForVersionUpgrade {
   /**
    * Barrier is usually started by the leader.
-   * @param version - for which the barrier is started.
-   * @param processorsNames - list of processors available at the time of the JobModel generation.
    */
-  void start(String version, List<String> processorsNames);
+  void start();
```

Comment:
Yeah. it can still be prepare() if you don't like prepareBarrier. Anyway, I will let you make the final call :)